### PR TITLE
feat(crons): Add `unassigned` to owner filter

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/monitors/endpoints/organization_monitor_index.py [hunk 1]
```diff
@@ -187,6 +188,14 @@ def get(self, request: Request, organization: Organization) -> Response:
             sort_fields = ["environment_status_ordering", "-last_checkin_monitorenvironment"]
         elif sort == "name":
             sort_fields = ["name"]
+        elif sort == "owner":
+            queryset = queryset.annotate(
+                owner_ordering=Case(
+                    When(Q(owner_user_id=None) & Q(owner_team_id=None), then=Value(0)),
+                    default=(F("owner_user_id"), F("owner_team_id")),
```
Comment (wedamija) [inline src/sentry/monitors/endpoints/organization_monitor_index.py:195]: Is the idea here that we want to sort owned things above unowned? Should the order actually be based on ids? That'd look fairly random. Maybe we want the default values to be `1` instead, so we show owned first, ordered by name, then unowned, ordered by name

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/organization_monitor_index.py:195]: oops sorry, this was half done lol

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/organization_monitor_index.py:195]: I did want to try and sort monitors on the same team / user together

Comment (wedamija) [reply src/sentry/monitors/endpoints/organization_monitor_index.py:195]: That's going to be a bit painful cross-db 😬 At least, it'll be hard to make it look alphabetical

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/organization_monitor_index.py:195]: this is true 😢

File: src/sentry/monitors/endpoints/organization_monitor_index.py [hunk 2]
```diff
@@ -203,9 +203,11 @@ def get(self, request: Request, organization: Organization) -> Response:
         if owners:
             owners = set(owners)
 
-            # Remove special 'myteams' from owners, this can't be parsed as an ActorTuple
-            myteams = ["myteams"] if "myteams" in owners else []
+            # Remove special values from owners, this can't be parsed as an ActorTuple
+            include_myteams = "myteams" in owners
             owners.discard("myteams")
```
Comment (wedamija) [inline src/sentry/monitors/endpoints/organization_monitor_index.py:208]: Nit: `include_myteams = owners.pop("myteams", None) is not None`?

File: src/sentry/monitors/endpoints/organization_monitor_index.py [hunk 3]
```diff
@@ -203,9 +203,11 @@ def get(self, request: Request, organization: Organization) -> Response:
...
+            include_unassigned = "unassigned" in owners
+            owners.discard("unassigned")
```
Comment (wedamija) [inline src/sentry/monitors/endpoints/organization_monitor_index.py:210]: Can do the same

**Review by wedamija — APPROVED** [review]

**Pull request merged**